server/index.js

```diff
@@ -69,6 +122,71 @@ app.post('/api/uploads', uploadsMiddleware, (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.post('/api/uploads/ratings/sign-up', uploadsMiddleware, (req, res, next) => {
+  const { ratedWho } = req.body;
+  if (!ratedWho) {
+    throw new ClientError(400, 'Who is being rated is a required field');
+  }
+  const sql = `
+    insert into "ratings" ("whoRated", "ratedWho", "rating")
+
+    SELECT 1 as "whoRated", "ratedWho", 5 as "rating"
+    from
+    (SELECT "username",
+      "accountId" as "ratedWho"
+    from "accounts"
+    where "username" = $1) as "a"
+    returning *;
+  `;
+  const params = [ratedWho];
+  db.query(sql, params)
+    .then(result => {
+      res.json(result.rows[0]);
+    })
+    .catch(err => next(err));
+});
+
+app.post('/api/uploads/ratings', uploadsMiddleware, (req, res, next) => {
+  let { whoRated, ratedWho, rating } = req.body;
+  rating = Number(rating);
+  if (!whoRated || !ratedWho) {
+    throw new ClientError(400, 'Who rated, who is being rated. and rating are required fields');
+  } else if (rating < 0 || rating > 10) {
+    throw new ClientError(400, 'Rating is outside of range');
+  }
+  const sql = `
+    insert into "ratings" ("whoRated", "ratedWho", "rating")
+
+    SELECT "whoRated", "ratedWho", $3 as "rating"
+    from
+    (SELECT "username",
+      "accountId" as "whoRated",
+      1 as "joinId"
+    from "accounts"
+    where "username" = $1) as "a"
+
+    join
+
+    (SELECT "username",
+      "accountId" as "ratedWho",
+      1 as "joinId"
+    from "accounts"
+    where "username" = $2) as "b"
+
+    using ("joinId")
+    returning *
+    ;
+  `;
```

Comment:
This SQL seems very convoluted, can we rework it to something like this? Can the client send us more convenient information?

```sql
insert into "ratings" ("whoRated", "ratedWho", "rating")
values (1, $1, $2)
returning *
```